src/rust/engine/src/nodes.rs

```diff
@@ -290,6 +291,33 @@ impl Select {
               }))
           .to_boxed(),
       ]
+    } else if self.product() == &context.core.types.process_result {
+      let value = externs::val_for_id(self.subject.id());
+      let mut env: BTreeMap<String, String> = BTreeMap::new();
+      let env_var_parts = externs::project_multi_strs(&value, "env");
+      // TODO: Error if env_var_parts.len() % 2 != 0
+      for i in 0..(env_var_parts.len() / 2) {
+        env.insert(
+          env_var_parts[2 * i].clone(),
+          env_var_parts[2 * i + 1].clone(),
+        );
+      }
+      let request = process_executor::ExecuteProcessRequest {
+        argv: externs::project_multi_strs(&value, "argv"),
+        env: env,
+      };
+      // TODO: this should run off-thread, and asynchronously
```

Comment:
This should be requesting the `Node` that invokes the process, rather than doing it directly. Can leave as a TODO if you'd like.